app/src/test/java/com/sdp/movemeet/modelsTest/UserDatabaseTest.java

```diff
@@ -0,0 +1,45 @@
+package com.sdp.movemeet.modelsTest;
+
+import com.sdp.movemeet.utility.UserDatabase;
+
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+public class UserDatabaseTest {
+    @Test
+    public void DatabaseConstructorIsCorrect() throws IOException {
+        UserDatabase db = new UserDatabase();
+
+        db.cleanFile();
+    }
+
+    @Test
+    public void AddingUsersIsCorrect() throws IOException {
+        UserDatabase db = new UserDatabase();
+
+        db.addUser("hello", "world");
+        db.addUser("hello", "world");
+        assertEquals(true, db.isInUsers("hello"));
+        assertEquals(false, db.isInUsers("goodbye"));
+    }
+
+    @Test
+    public void SavingAndLoadingIsCorrect() throws IOException {
+        UserDatabase db = new UserDatabase();
+
+        db.addUser("hello", "world");
+
+        db.writeToFile();
```

Comment:
Given that nothing is tested here, I assume that this doesn't really work yet ? How do we know that the map of users written corresponds to the one read ?

Reply:
It does work, but I haven't added it to the test suite. 
During the next sprint, I am going to enrich the test suite so that it covers more cases and tests this class thoroughly.